playbooks/operationalizing/ingress.adoc

```diff
@@ -0,0 +1,202 @@
+
+[source]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: null
+  labels:
+    app: postgresql-persistent
+    template: postgresql-persistent-template
+  name: postgresql-ingress
+spec:
+  deprecatedPublicIPs:
+  - 10.9.54.100
+  externalIPs:
+  - 10.9.54.100
+  ports:
+  - nodePort: 32439 <1>
+    port: 5432
+    protocol: TCP
+    targetPort: 5432
+  selector:
+    name: postgresql
+  sessionAffinity: None
+  type: LoadBalancer
+----
+<1> Automatically assigned port
+
+A PostgreSQL client can now be configured to connect directly to any node using the value of the assigned _nodePort_.
+
+=== Configuring the Service to be Highly Available
+
+Instead of connecting directly to individual node's, a more robust strategy for providing access services configured with external ip addresses is to utilize one of OpenShift's link:https://docs.openshift.com/container-platform/latest/admin_guide/high_availability.html[high availability strategies] by deploying ipfailover router. This allows cluster administrators the flexibility of defining the ingress points within a cluster, and ensuring the service is readily available. 
+
+NOTE: Nodes that will have ipfailover routers deployed to them must be in the same Layer 2 switching domain for ARP broadcasts to communicate with switches the appropriate port the destination should flow.
+
+CAUTION: Only 254 addresses can be allocated in a single VLAN due to the restrictions placed by the Virtual Router Redundancy Protocol (VRRP) which is used by the ipfailover router.
+
+Since routers require privileged access to the host they are deployed to facilitate port binding, a separate service account called _ipfailover_ will be used to run the pod containing the router. 
+
+Preconfigure the service account to be a member of the _privileged_ link:https://docs.openshift.com/container-platform/latest/admin_guide/manage_scc.html[security context constraint] by executing the following command:
+
+[source]
+----
+oadm policy add-scc-to-user privileged system:serviceaccount:default:ipfailover
+----
+
+Next, multicast must be enabled on each node the router will be deployed to in order for it to receive traffic for 224.0.0.18 (the VRRP multicast IP). Depending on your environment, enable multicast on each of the nodes by executing the following command
+
+[source]
+----
+sudo /sbin/iptables -I INPUT -i <interface> -d 224.0.0.18/32 -j ACCEPT
+----
+
+NOTE: Replace the <interface> variable with the applicable network interface receiving traffic
+
+Finally, deploy the ipfailover router to monitor postgresql listening on node port 32439 and the external IP address as defined in the `postgresql-ingress` service:
+
+[source]
+----
+oadm ipfailover ipf-ha-postgresql \
+    --replicas=1 <1> --selector="app-type=postgresql" <2> \ 
```

Comment:
I don't think I like the implied suggestion of a nodeSelector having a value so specific to a type of service. I would think we should suggest some kind of best practice of designating a special set of infranodes to handle ingress traffic. In my mind, there's no real reason this couldn't just be the same ones that run router/registry. I would at minimum change the selector here to be more generic, and then include some callout explaining the use of infranodes here.